### PR TITLE
More uses of ArgumentOutOfRangeException throw helpers

### PR DESCRIPTION
**Pull request by halgab**
Related: #9726

So far I left alone quite a few instances of manual `ArgumentOutOfRangeException` creation and throwing. Those are the ones in property setter.

Their error messages mention both the `value` parameter and the name of the propery itself. I we want to get rid of custom error messages, we need to take care of them, but if we do so we have to choose between the two names.

So far in this PR I only touched two properties to show our options:

1. Use `value` in the error message
2. Use the property name in the error message
3. Do nothing

Once a decision has been made, I'll use this PR to enforce it across codebase. Then we will be in a position to close the issue above

@lonitra 

## Regression? 

- No

## Risk

- Low: some error messages will change

# Test methodology 

- CI

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/10423)

### CODE REVIEW

**Comment by lonitra** [comment]
Personally I would prefer option 1 where I'd be able to see what the exact value is to blame and can investigate further by looking at stack trace to determine which property versus option 2 I'm not so sure would have all the information by just looking at message + stack trace, but I will also let @JeremyKuhne give his thoughts as well.

**Comment by RussKie** [comment]
Where you do know the argument name - you should use it.

File: src/System.Windows.Forms/src/System/Windows/Forms/Controls/Buttons/ButtonBase.cs [hunk 1]
```diff
@@ -478,10 +478,7 @@ public int ImageIndex
         }
         set
         {
-            if (value < ImageList.Indexer.DefaultIndex)
-            {
-                throw new ArgumentOutOfRangeException(nameof(value), value, string.Format(SR.InvalidLowBoundArgumentEx, nameof(ImageIndex), value, ImageList.Indexer.DefaultIndex));
-            }
+            ArgumentOutOfRangeException.ThrowIfLessThan(value, ImageList.Indexer.DefaultIndex);
```
[the remark this answers is not in the document]
Comment (JeremyKuhne) [reply src/System.Windows.Forms/src/System/Windows/Forms/Controls/Buttons/ButtonBase.cs:481]: You should be able to use `nameof(ImageIndex)`?

[the remark this answers is not in the document]
Comment (halgab) [reply src/System.Windows.Forms/src/System/Windows/Forms/Controls/Buttons/ButtonBase.cs:481]: Yes, but then there would be no mention of `value`. That's exactly option 2. Do you favor that pattern then?

**Comment by halgab** [comment]
Has the team made a decision on this? My take is that most of the time the argument name doesn't have much value in setters as you can't choose it, but really all solutions have pros and cons here.

File: src/System.Windows.Forms/src/System/Windows/Forms/Controls/Buttons/ButtonBase.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (lonitra) [reply src/System.Windows.Forms/src/System/Windows/Forms/Controls/Buttons/ButtonBase.cs:481]: @JeremyKuhne gentle ping for a decision on this.

[the remark this answers is not in the document]
Comment (JeremyKuhne) [reply src/System.Windows.Forms/src/System/Windows/Forms/Controls/Buttons/ButtonBase.cs:481]: Sorry about that. Go with simpler. What you have here is the best option.

``` C#
 ArgumentOutOfRangeException.ThrowIfLessThan(value, ImageList.Indexer.DefaultIndex);
```

File: src/System.Windows.Forms/src/System/Windows/Forms/Controls/RichTextBox/RichTextBox.cs [hunk 2]
```diff
@@ -1508,13 +1491,8 @@ public unsafe float ZoomFactor
...
         set
         {
-            if (value <= 0.015625f || value >= 64.0f)
-            {
-                throw new ArgumentOutOfRangeException(
-                    nameof(value),
-                    value,
-                    string.Format(SR.InvalidExBoundArgument, nameof(ZoomFactor), value, 0.015625f, 64.0f));
-            }
+            ArgumentOutOfRangeException.ThrowIfLessThanOrEqual(value, 0.015625f);
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/Controls/RichTextBox/RichTextBox.cs:1494]: `CompareTo` treats NaN as a special case, does not equate it to 0.015625f. I would keep the old condition and exception for `if (value <= 0.015625f )`

File: src/System.Windows.Forms/src/System/Windows/Forms/Controls/ProgressBar/ProgressBar.cs [hunk 3]
```diff
@@ -463,7 +454,7 @@ public int Value
             {
                 if (value < _minimum || value > _maximum)
                 {
-                    throw new ArgumentOutOfRangeException(nameof(value), value, string.Format(SR.InvalidBoundArgument, nameof(Value), value, "'minimum'", "'maximum'"));
```
[the remark this answers is not in the document]
Comment (halgab) [reply src/System.Windows.Forms/src/System/Windows/Forms/Controls/ProgressBar/ProgressBar.cs:466]: I figured this one was a bit different, as it explicits the properties to check to understand the boundaries. But I can change this (and other similar instances) as well if you prefer

File: src/System.Windows.Forms/src/System/Windows/Forms/Controls/RichTextBox/RichTextBox.cs (hunk 2 above)
Comment (halgab) [reply src/System.Windows.Forms/src/System/Windows/Forms/Controls/RichTextBox/RichTextBox.cs]: I chose to special case NaN to keep the throw helper. To have a good error message, the alternative would be to keep around `SR.InvalidExBoundArgument`, and one of the points of this change is to get rid of such redundant messages

File: src/System.Windows.Forms/src/System/Windows/Forms/Controls/ProgressBar/ProgressBar.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (Tanya-Solyanik) [reply src/System.Windows.Forms/src/System/Windows/Forms/Controls/ProgressBar/ProgressBar.cs:466]: Sounds good!

**Review by Tanya-Solyanik — APPROVED** [review]
Looks good, thanks!

**Pull request merged**